### DISCUSSION TITLE
Does a Trezor-connected Metamask wallet share the same secret phrase/mnemonic as other Metamask accounts?

### DISCUSSION THREAD

**Discussion by dalteko** [Cyfrin/foundry-full-course-cu#132] [Q&A] [answered]
I am aware that even if I hit add account on my metamask (or other ETH wallet) I will get a new private key, but it will share the same secret phrase/mnemonic of the other accounts generated in that metamask (or other ETH wallet).

This is part of the .env pledge, and I'm trying to follow it! But I've linked my Trezor to my Metamask browser extension wallet, so I'm wondering if this specific part still holds true: it will share the same secret phrase/mnemonic of the other accounts generated in that metamask (or other ETH wallet). 

Do the accounts I make using Anvil/Ganache share the same secret phrase/mnemonic as my Trezor-connected MetaMask wallet?

If it does, I'll obviously have to re-do lesson 6 with a new wallet! Which is fine, but I just wanted to confirm first.

**Comment by cromewar** [comment, marked as answer]
Hello @dalteko 
Thanks for your question, the usage of the mnemonic phrase is thanks to something called [Bip39](https://vault12.com/securemycrypto/crypto-security-basics/what-is-bip39/) this is the standard we use to generate mnenomic phrases, I strongly suggest you to give a look to the article on the link.

MetaMask, when linked to Trezor, does not have access to your Trezor's mnemonic. Instead, MetaMask asks Trezor to generate public keys for accounts using the HD (Hierarchical Deterministic) Wallets protocol. Essentially, it's like asking Trezor to "show" MetaMask a bunch of accounts that belong to your Trezor device without revealing the actual mnemonic to MetaMask.

Ganache, on the other hand, generates a new mnemonic every time you start it, and the accounts associated with that mnemonic have no relation to your Trezor or MetaMask mnemonics unless you specifically configure Ganache to use your own mnemonic, which is not recommended for security reasons.

So, while the accounts generated from MetaMask when connected to Trezor and the accounts directly on Trezor will share the same mnemonic, the accounts generated in Anvil/Ganache won't share the same mnemonic with your Trezor unless specifically configured.

**Reply by dalteko** [reply]
Appreciate it! This means that I can use the private key from another non-Trezor connected Metamask account in the .env file :D
